freshmaker/kojiservice.py

```diff
@@ -407,6 +412,29 @@ def get_bundle_csv(self, build_nvr):
         with zipfile.open(csv_files[0]) as f:
             return yaml.safe_load(f)
 
+    @region.cache_on_arguments()
+    def get_modulemd(self, build_nvr):
+        """
+        Get Modulemd.ModuleStreamV2 instance through build NVR
+        which can be used to get module's name, stream, etc.
+
+        :param str build_nvr: NVR of module build.
+        :return: Modulemd.ModuleStreamV2 instance.
+        """
+        mmd = None
+        build = self.get_build(build_nvr)
+        try:
+            modulemd_str = build["extra"]["typeinfo"]["module"]["modulemd_str"]
+        except KeyError:
+            return None
+
+        if modulemd_str:
+            try:
+                _, mmd = Modulemd.read_packager_string(modulemd_str, None, None)
+            except Exception as e:
+                log.error("Read packager string failed for NVR %s: %s", build_nvr, str(e))
+        return mmd
```

Comment:
```suggestion
        try:
            _, mmd = Modulemd.read_packager_string(modulemd_str, None, None)
            return mmd
        except Exception as e:
            log.error("Read packager string failed for NVR %s: %s", build_nvr, str(e))
            return None
```

Reply:
I know some convention says return immediately, and some says one return point.  I don't have strong opinion to use one not the other.

Reply:
Updated